hedera-node/hedera-mono-service/src/test/java/com/hedera/services/store/contracts/precompile/TransferPrecompilesTest.java

```diff
@@ -1402,12 +1409,123 @@ void gasRequirementReturnsCorrectValueForTransferSingleToken() {
+        given(worldUpdater.permissivelyUnaliased(any()))
+                .willAnswer(invocationOnMock -> invocationOnMock.getArgument(0));
+        given(syntheticTxnFactory.createCryptoTransfer(any()))
+                .willReturn(
+                        TransactionBody.newBuilder()
+                                .setCryptoTransfer(CryptoTransferTransactionBody.newBuilder()));
+        given(worldUpdater.permissivelyUnaliased(any()))
+                .willAnswer(invocationOnMock -> invocationOnMock.getArgument(0));
+        transferPrecompile
+                .when(() -> decodeCryptoTransferV2(eq(input), any()))
+                .thenReturn(CRYPTO_TRANSFER_HBAR_ONLY_WRAPPER);
+
+        feeBuilder
+                .when(() -> getTinybarsFromTinyCents(any(), anyLong()))
+                .thenReturn(TEST_CRYPTO_TRANSFER_MIN_FEE);
+
+        given(dynamicProperties.isAtomicCryptoTransferEnabled()).willReturn(true);
+
+        subject.prepareFields(frame);
+        subject.prepareComputation(input, a -> a);
+        var minimumFeeInTinybars = subject.getPrecompile().getMinimumFeeInTinybars(timestamp);
+
+        // then
+        assertEquals(TEST_CRYPTO_TRANSFER_MIN_FEE, minimumFeeInTinybars);
+
+        feeBuilder.close();
+    }
+
+    @Test
+    void minimumFeeInTinybarsHbarFungibleCryptoTransfer() {
+        var feeBuilder = Mockito.mockStatic(FeeBuilder.class);
+
+        // given
+        givenMinFrameContext();
+        Bytes input = Bytes.of(Integers.toBytes(ABI_ID_CRYPTO_TRANSFER_V2));
+        given(infrastructureFactory.newSideEffects()).willReturn(sideEffects);
+        given(worldUpdater.permissivelyUnaliased(any()))
+                .willAnswer(invocationOnMock -> invocationOnMock.getArgument(0));
+        given(syntheticTxnFactory.createCryptoTransfer(any()))
+                .willReturn(
+                        TransactionBody.newBuilder()
+                                .setCryptoTransfer(CryptoTransferTransactionBody.newBuilder()));
+        given(worldUpdater.permissivelyUnaliased(any()))
+                .willAnswer(invocationOnMock -> invocationOnMock.getArgument(0));
+        transferPrecompile
+                .when(() -> decodeCryptoTransferV2(eq(input), any()))
+                .thenReturn(CRYPTO_TRANSFER_HBAR_FUNGIBLE_WRAPPER);
+
+        feeBuilder
+                .when(() -> getTinybarsFromTinyCents(any(), anyLong()))
+                .thenReturn(TEST_CRYPTO_TRANSFER_MIN_FEE);
+
+        given(dynamicProperties.isAtomicCryptoTransferEnabled()).willReturn(true);
+
+        subject.prepareFields(frame);
+        subject.prepareComputation(input, a -> a);
+        var minimumFeeInTinybars = subject.getPrecompile().getMinimumFeeInTinybars(timestamp);
+
+        // then
+        // 1 for hbars and 1 for fungible tokens
+        assertEquals(2 * TEST_CRYPTO_TRANSFER_MIN_FEE, minimumFeeInTinybars);
+
+        feeBuilder.close();
+    }
+
+    @Test
```

Comment:
Could you add  also test case for hbar transfer + fungible transfer + non fungible?

Reply:
done